From 001cdfec7109774269e1cb3adeff06654ed5ca42 Mon Sep 17 00:00:00 2001
From: Shawn Lin <shawn.lin@rock-chips.com>
Date: Thu, 24 May 2018 14:31:00 +0800
Subject: [PATCH] mmc: core: Allow adding software debounce via
 mmc_gpiod_request_cd()

mmc_gpio_request_cd() allow host drivers to add bigger software
debounce from the drivers themself, but we might allow the same
behaviour from the fwnode.

Change-Id: Icc994aa9b58b0b8836a2fb60ac1a87ac3c7e676a
Signed-off-by: Shawn Lin <shawn.lin@rock-chips.com>
---
 drivers/mmc/core/slot-gpio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/core/slot-gpio.c b/drivers/mmc/core/slot-gpio.c
index 9355140d8bb9..5c6e83bcce66 100644
--- a/drivers/mmc/core/slot-gpio.c
+++ b/drivers/mmc/core/slot-gpio.c
@@ -248,7 +248,7 @@ int mmc_gpiod_request_cd(struct mmc_host *host, const char *con_id,
 	if (debounce) {
 		ret = gpiod_set_debounce(desc, debounce);
 		if (ret < 0)
-			return ret;
+			ctx->cd_debounce_delay_ms = debounce;
 	}
 
 	if (gpio_invert)
-- 
2.35.3

